browser/ui/views/frame/vertical_tab_strip_region_view.cc

```diff
@@ -524,9 +525,14 @@ class VerticalTabStripRegionView::HeaderView : public views::View {
       return;
     }
 
+    // Note: We need to copy the vector of children here so we don't modify the
+    // vector while iterating.
+    std::vector<views::View*> children_copy;
+    base::ranges::copy(children(), std::back_inserter(children_copy));
+
     // View::ReorderChildView() didn't work for us. So remove child views and
     // add them again.
-    base::ranges::for_each(children(),
+    base::ranges::for_each(children_copy,
```

Comment:
without this, I was encountering crashes - I've made a separate PR for this as it's not really related, and I don't know if this PR will land :laughing: 

https://github.com/brave/brave-core/pull/25737